docs/data/joy/components/select/SelectFieldDemo.js

```diff
@@ -9,20 +9,26 @@ export default function SelectFieldDemo() {
   const [value, setValue] = React.useState('dog');
   return (
     <Box sx={{ width: 240 }}>
-      <FormLabel>Favorite pet</FormLabel>
+      <FormLabel
+        id="select-field-demo-label"
+        htmlFor="select-field-demo-button"
+        sx={{ mb: 0.25 }}
+      >
+        Favorite pet
+      </FormLabel>
       <Select
         defaultValue="dog"
         componentsProps={{
           button: {
+            id: 'select-field-demo-button',
             // screen readers will announce "Favorite pet, dog selected" when the select is focused.
-            'aria-label': `Favorite pet, ${value} selected.`,
+            'aria-labelledby': 'select-field-demo-label select-field-demo-button',
             // and this in the next sentence.
             'aria-describedby': 'select-field-demo-helper',
           },
         }}
         value={value}
         onChange={setValue}
-        sx={{ mt: 0.25 }}
```

Comment:
I think that it should be owned by the label, if a developer clone this demo and remove the label, this margin should be gone too.